packages/astro/src/cli/check/index.ts

```diff
@@ -1,386 +1,39 @@
-	ExitWithError,
-	/**
-	 * The consumer should not terminate the operation
-	 */
-	Listen,
-}
-
-const ASTRO_GLOB_PATTERN = '**/*.astro';
-
-/**
- * Checks `.astro` files for possible errors.
- *
- * If the `--watch` flag is provided, the command runs indefinitely and provides diagnostics
- * when `.astro` files are modified.
- *
- * Every time an astro files is modified, content collections are also generated.
- *
- * @param {CheckPayload} options Options passed {@link AstroChecker}
- * @param {Flags} options.flags Flags coming from the CLI
- * @param {LogOptions} options.logging Logging options
- */
-export async function check({ logging, flags }: CheckPayload): Promise<AstroChecker | undefined> {
-	if (flags.help || flags.h) {
-		printHelp({
-			commandName: 'astro check',
-			usage: '[...flags]',
-			tables: {
-				Flags: [
-					['--watch', 'Watch Astro files for changes and re-run checks.'],
-					['--help (-h)', 'See all available flags.'],
-				],
-			},
-			description: `Runs diagnostics against your project and reports errors to the console.`,
-		});
-		return;
-	}
-
-	const settings = await loadSettings({ cmd: 'check', flags, logging });
-	if (!settings) return;
-
-	const checkFlags = parseFlags(flags);
-	if (checkFlags.watch) {
-		info(logging, 'check', 'Checking files in watch mode');
-	} else {
-		info(logging, 'check', 'Checking files');
-	}
-
-	const { syncCli } = await import('../../core/sync/index.js');
-	const root = settings.config.root;
-	const require = createRequire(import.meta.url);
-	const diagnosticChecker = new AstroCheck(
-		root.toString(),
-		require.resolve('typescript/lib/tsserverlibrary.js', {
-			paths: [root.toString()],
-		})
-	);
-
-	return new AstroChecker({
-		syncCli,
-		settings,
-		fileSystem: fs,
-		logging,
-		diagnosticChecker,
-		isWatchMode: checkFlags.watch,
-	});
-}
-
-type CheckerConstructor = {
-	diagnosticChecker: AstroCheck;
-
-	isWatchMode: boolean;
-
-	syncCli: (settings: AstroSettings, options: SyncOptions) => Promise<ProcessExit>;
-
-	settings: Readonly<AstroSettings>;
-
-	logging: Readonly<LogOptions>;
-
-	fileSystem: typeof fs;
-};
-
-/**
- * Responsible to check files - classic or watch mode - and report diagnostics.
- *
- * When in watch mode, the class does a whole check pass, and then starts watching files.
- * When a change occurs to an `.astro` file, the checker builds content collections again and lint all the `.astro` files.
- */
-export class AstroChecker {
-	readonly #diagnosticsChecker: AstroCheck;
-	readonly #shouldWatch: boolean;
-	readonly #syncCli: (settings: AstroSettings, opts: SyncOptions) => Promise<ProcessExit>;
-
-	readonly #settings: AstroSettings;
-
-	readonly #logging: LogOptions;
-	readonly #fs: typeof fs;
-	#watcher?: FSWatcher;
-
-	#filesCount: number;
-	#updateDiagnostics: NodeJS.Timeout | undefined;
-
-	constructor({
-		diagnosticChecker,
-		isWatchMode,
-		syncCli,
-		settings,
-		fileSystem,
+export async function check({ flags, logging }: CheckOptions) {
```

Comment:
Also a merge hint when rebasing `next` to align with `main`. The logging would need to be derived like so:

```js
const logging = createLoggingFromFlags(flags);
```